src/c3py/history.py

```diff
@@ -61,6 +61,20 @@ def __init__(self, data: dict[str, list[Operation]]):
         for process, ops in data.items():
             for i in range(len(ops) - 1):
                 self.poset.order_try(f"{process}.{i + 1}", f"{process}.{i + 2}")
+    
+    # co = (po U wr)^+
+    def causal_order(self):
```

Comment:
I don't want to add this `causal_order` method on `History` because this method only makes sense for histories of read/write memory. I think the easiest approach would be to make a subclass `WRMemoryHistory` that extends `History` and move this method there.